src/idom/mypy.py

```diff
@@ -0,0 +1,51 @@
+from typing import Callable, Final
+
+from mypy.errorcodes import ErrorCode
+from mypy.nodes import ArgKind
+from mypy.plugin import FunctionContext, Plugin
+from mypy.types import CallableType, Instance, Type
+
+
+KEY_IN_RENDER_FUNC: Final = ErrorCode(
+    "idom-key-in-render-func",
+    "Component render function has reserved 'key' parameter",
+    "IDOM",
+)
+
+
+class MypyPlugin(Plugin):
+    """MyPy plugin for IDOM"""
+
+    def get_function_hook(self, fullname: str) -> Callable[[FunctionContext], Type]:
+        if fullname == "idom.core.component.component":
```

Comment:
Are we allowed to do an `from idom import component` and then have this check be `fullname == f"{component.__module}.{component.__name__}"`? 

Would improve refactorability.

Reply:
Yes, however, if we decide to change the path `ido.core.component.component` in the future, this plugin will break silently.

I would recommend using python imports to generate that path, that way refactoring libraries such as [`rope`](https://github.com/python-rope/rope) can automatically update these paths. Or at the very least, this plugin would create a visible exception to demonstrate that things are broken.